### PR TITLE
[5.4] Add "contains" and "not_contains" validation rules

### PR DESCRIPTION
**Pull request by mcordingley**
Adding the "not_contains" validation rule, since it is awkward to implement using existing validation rules. Adding the "contains" rule for symmetry.

### CODE REVIEW

File: src/Illuminate/Validation/Validator.php [hunk 1]
```diff
@@ -1828,6 +1828,32 @@ protected function validateRegex($attribute, $value, $parameters)
...
+     * Validate that an attribute contains none of a list of list of substrings.
+     *
+     * @param $attribute
+     * @param $value
+     * @param $parameters
+     * @return bool
+     */
+    protected function validateNotContains($attribute, $value, $parameters)
+    {
+        return ! str_contains($value, $parameters);
```
Comment (srmklive) [inline src/Illuminate/Validation/Validator.php]: Can't we also do something like this here:

    return ! $this->validateContains($attribute, $value, $parameters);

**Comment by taylorotwell** [comment]
Interesting. What was your use case for this?

**Comment by mcordingley** [comment]
Taking down a password for a network device. There are a few different characters that would cause it to choke, so I have a local version of this enforcing the black-list.

File: src/Illuminate/Validation/Validator.php (hunk 1 above)
Comment (mcordingley) [inline src/Illuminate/Validation/Validator.php]: Sure, but what I have is simpler.

**Comment by taylorotwell** [comment]
I kinda think the regex rule could just be used for this for now.

**Comment by mcordingley** [comment]
For "contains", definitely. Putting together a regex for a set of patterns not to be matched is quite a bit harder.

**Pull request closed without merging**